src/player/player_state_machine.gd

```diff
@@ -0,0 +1,73 @@
+extends "res://src/state/finite_state_machine.gd"
+
+class_name PlayerStateMachine
+
+func _init(p):
+	add_state("idle")
+	add_state("walk")
+	add_state("jump")
+	add_state("fall")
+	parent = p
+
+func _state_logic(delta):
+	match state:
+		states.idle:
+			parent.apply_idle(delta)
+		states.walk:
+			parent.apply_move(delta)
+		states.jump:
+			parent.apply_jump(delta)
+			parent.apply_move(delta)
+		states.fall:
+			parent.apply_fall(delta)
+
+func _get_transition(delta):
```

Comment:
Need to make more generic because number of states will grow a lot, but ok for now.

Reply:
you fucking bitch